File: jano_motors_app/src/app/Interfaces/booking.ts
import { Car } from "./car";

export interface BookingGet {
    id: number;
    userId: number;
    carId: number;
    startDate: Date;
    endDate: Date;
    totalPrice: Float32Array;
    status: string;
}

export interface Booking {
    id: number;
    userId: number;
    car: Car;
    startDate: Date;
    endDate: Date;
    totalPrice: Float32Array;
    status: string;
}

export interface BookingPost {
    userId: number;
    carId: number;
    startDate: string;
    endDate: string;
    totalPrice: number;
    status: string;
}

File: JanoMotorsApi/JanoMotorsApi/Models/Booking.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace JanoMotorsApi.Models
{
    public class Booking
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public required int UserId { get; set; }
        public required int CarId { get; set; }
        public required DateTime StartDate { get; set; }
        public required DateTime EndDate { get; set; }
        public required decimal TotalPrice { get; set; }
        public required string Status { get; set; }

        [JsonIgnore]
        public User? User { get; set; }
        [JsonIgnore]
        public Car? Car { get; set; }
        [JsonIgnore]
        public ICollection<PaymentDoc>? PaymentDocs { get; set; }
    }
}
